src/cpp/rtps/transport/TCPChannelResourceBasic.h

```diff
@@ -81,6 +82,10 @@ class TCPChannelResourceBasic : public TCPChannelResource
         return socket_;
     }
 
+    void update_channel(
+            const std::shared_ptr<asio::ip::tcp::socket> socket,
+            const eConnectionStatus status);
```

Comment:
This method could receive another channel, and take all the required information from there